src/plugins/discover/public/application/main/components/layout/discover_main_content.tsx

```diff
@@ -145,7 +151,16 @@ export const DiscoverMainContent = ({
                 trackUiMetric={trackUiMetric}
               />
             </>
-          )}
+          ) : null}
+          {viewMode === VIEW_MODE.PATTERN_LEVEL ? (
+            <PatternAnalysisTab
+              dataView={dataView}
+              stateContainer={stateContainer}
+              onAddFilter={() => setDiscoverViewMode(VIEW_MODE.DOCUMENT_LEVEL)}
```

Comment:
Are there any checks we can add here to prevent this happening if you select a lot of patterns at once to filter in / out (here I tried to filter out 50 patterns in one go):

<img width="662" alt="Screenshot 2024-05-15 at 12 30 52" src="https://github.com/elastic/kibana/assets/7405507/9edc6dfc-1574-4190-b66f-4f528a0291c5">

Reply:
We could try to predict the size of the url as the user is selecting patterns. However, this will not stop subsequent non-pattern related filters or URL stored settings from tipping it over the limit.
It'd just mean that as the user is selecting patterns we could show a warning that it might be over the limit.

The number of patterns which cause a problem depends on the size of their pattern keys. The user may have selected only a few (50) patterns but the keys are all very long and so might cause a problem compared to a pile of patterns with short keys.